keyboards/misterknife/knife66_iso/keymaps/default/keymap.c

```diff
@@ -0,0 +1,45 @@
+/*
+Copyright 2020 Thierry Kleist
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#include QMK_KEYBOARD_H
+
+// Defines names for use in layer keycodes and the keymap
+enum layer_names {
+    _BASE,
+    _FN
+};
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [_BASE] = LAYOUT_all(
+    KC_GESC,   KC_1,     KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS,  KC_EQL,   KC_DEL, KC_BSPC,  KC_HOME,
+    KC_TAB,    KC_Q,     KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC,  KC_RBRC,  KC_ENT,           KC_END,
+    KC_LCTL,   KC_A,     KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,            KC_BSLS,          KC_PGUP,
+    KC_LSFT,   KC_GRAVE, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH,  KC_RSFT,  KC_UP,            KC_PGDN,
+    KC_LCTL,             KC_LGUI, KC_LALT,          LT(1, KC_SPC),    KC_SPC,           KC_RALT, MO(1),   KC_RCTL,  KC_LEFT,  KC_DOWN, KC_RGHT),
+
+  [_FN] = LAYOUT_all(
+    KC_TRNS,   KC_F1,    KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,    KC_F8,    KC_F9,    KC_F10,   KC_F11,   KC_F12,   KC_TRNS, KC_TRNS, RGB_TOG,
+    KC_TRNS,   KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_MPLY,  KC_MPRV,  KC_MNXT,  KC_TRNS,          RGB_MOD,
+    KC_TRNS,   KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_LEFT, KC_DOWN,  KC_UP,    KC_RIGHT, KC_TRNS,  KC_TRNS,            KC_INS,           RGB_SPI,
+    KC_CAPS,   KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_VOLU,          RGB_SPD,
+    KC_TRNS,             KC_TRNS, KC_TRNS,          KC_TRNS,          KC_TRNS,            KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_MUTE,  KC_VOLD, KC_TRNS ),
+};
+
+#ifdef RGBLIGHT_ENABLE
+void keyboard_post_init_user(void) {
+  rgblight_mode_noeeprom(10);
```

Comment:
It may be best to use the defined name here, as this may change based on what modes are enabled or not.

Reply:
> Just a heads up, we do reviews in our free time. That means it can take some time to get a review or response. More so because it's been quite a year. And because we're entering into the holiday season.
> 
> That said, if you have free time and are up for it, anyone can review PRs. The only thin is that it won't be colored (green/red), but it does help us out. But if you don't want to, or don't have the time to, don't worry it.

thanks for the explanation. I'm sorry if I sounded harsh. this was not my intent. 

I can help but I'm not a native in cpp or c. I am getting better at it but this could help me teach more I think. 
Where can I apply?

Reply:
No, you didn't sound harsh.  It was mostly a "heads up, this is why".  We do this because we love qmk.  But I'm sure you know how "time" goes.  Not enough hours in the day, week, month, etc. 

And basically, it's just using the entries from this chart:
https://docs.qmk.fm/#/feature_rgblight?id=effects-and-animations

And i think it's the rainbow swirl you're looking for  So:
```c
rgblight_mode_noeeprom(RGBLIGHT_MODE_RAINBOW_SWIRL+1);
```

Reply:
thanks! pushed